### DISCUSSION TITLE
Issue with plroute

### DISCUSSION THREAD

**Discussion by walls032** [smaret/astrochem#97] [Q&A]
Hi @smaret,

I was getting strange behavior when trying to run plroute from the command line after a successful astrochem run wherein I track the abundances of the following species:
    H, H(+), H2, O, C(+), C, CO, H2O, and O2
and trace formation/destruction routes via: trace_routes = 1

Specifically, I get the following for "H":
    $ plroute astrochem_output.h5 "H"
    $ plroute: error: can't find reaction 24071520 in <chemical network file>
where I have redacted the chemical network file, which has at most 133 reactions. This error occurs for the following of the species above: 
   H, C(+), and C

Having read the installation prerequisites as well as the issues section, I know plroute had similar strange behavior when h5py version 2.6.0 was used as opposed to version 2.5.0. However, I cannot find a stable version of h5py 2.5.0 available for use to test whether that is really the issue. 

Any recommendations for resolving/circumventing this problem?

Many thanks,
Levi Walls

**Comment by smaret** [comment]
Hi Levi,

Thanks for your message. Could you please send me the input files that you are using so I can reproduce this problem ?

**Reply by walls032** [reply]
Hi Sebastien,

Within the attached tar file are the following files:
- **input_n9p5.ini**: contains all the input parameters. Note the grain abundance is specified but unused by this network.
- **source_n9p5.mdl**: contains the properties of a single cell for times 1e-6 to 1e7 yrs
- **h2o_Ted_OSU_nograin_noD.chm**: contains gas-phase reactions using OSU 2009 reaction rate coefficient constants.

[astrochem_test_files.tar.gz](https://github.com/smaret/astrochem/files/14055516/astrochem_test_files.tar.gz)

Thank you for your help with this! 
If there is anything I can do to help, just let me know.

All the best,
Levi

**Comment by smaret** [comment]
Hi Levi,

Sorry it took me song long to answer, I've been swamped with other duties.
I can't reproduce the error you've encountered:

```
plabun astrochem_output.h5 "C(+)"
```

![plroute](https://github.com/smaret/astrochem/assets/95471/a3888727-1db0-4e67-991a-403d49197184)

I may indeed be related to `h5py` as you suggested. Which version are you using?  On my side:

```py
>>> import h5py
>>> h5py.__version__
'3.6.0'

```

**Reply by smaret** [reply]
Looking again at #50, the test suite should report an error if there is a problem with `h5py`. Do you get an error if you type 

```
make check
```

after the compilation?
